### PR TITLE
Tongwen11/multigpuvms

### PR DESCRIPTION
**Pull request by tongwen11**
Add the support to train on multiple GPU VMs along with
- Change the DataTransfer option value from "hdfs-mount" to "hdfs"
- Only copy the model file instead of the model folder which also contains checkpoint files
- Couple touch up changes

### CODE REVIEW

**Comment by elibarzilay** [comment]
@tongwen11 There are no changes to the ARM templates?  No argument for a number of GPU machines or something similar?

**Comment by elibarzilay** [comment]
I'm going to merge it and do the fixes myself for the things that I found in a different PR.

File: src/cntk-train/src/main/scala/BrainscriptBuilder.scala [hunk 1]
```diff
@@ -16,6 +16,7 @@ case class BrainScriptConfig(name: String, text: Seq[String])
 class BrainScriptBuilder {
 
   var modelName = "ModelOut"
+  var modelOutputDir = "Models"
 
   var inData: Option[InputData] = None
```
[the remark this answers is not in the document]
Comment (tongwen11) [reply src/cntk-train/src/main/scala/BrainscriptBuilder.scala]: Good catch. In 
def setRootDir(p: String): this.type = {
    outDir = p
    this
  }
outDir should be rootDir

[the remark this answers is not in the document]
Comment (tongwen11) [reply src/cntk-train/src/main/scala/BrainscriptBuilder.scala]: Yean, need a better default value for outDir.  I am thinking of /tmp/gpuworkingdir/CNTKLearner-outdir.

File: src/cntk-train/src/main/scala/BrainscriptBuilder.scala [hunk 2]
```diff
@@ -43,6 +44,11 @@ class BrainScriptBuilder {
     this
   }
 
+  def setModelOutputDir(n: String): this.type = {
```
[the remark this answers is not in the document]
Comment (tongwen11) [reply src/cntk-train/src/main/scala/BrainscriptBuilder.scala]: Yeah, forgot to change it to "p" which is used by other methods.

File: src/cntk-train/src/main/scala/CommandBuilders.scala [hunk 3]
```diff
@@ -178,30 +214,53 @@ class MPICommandBuilder(log: Logger,
...
+        for ((nodeName,numGPUs) <- nodeConfig) {
+          val remoteUser = s"$username@$nodeName"
+          printOutput(
+            Seq("scp", "-i", identity, "-r", "-o", "StrictHostKeyChecking=no",
+              fileInputPath, s"$remoteUser:$fileInputPath"))
+        }
+      }
+
+      // Validate ssh connection from the primary GPU machine to the rest in the list
+      if (nodeConfig.size > 1) {
```
[the remark this answers is not in the document]
Comment (tongwen11) [reply src/cntk-train/src/main/scala/CommandBuilders.scala]: You mean for ((nodeName, numGPUs) <- nodeConfig - pimaryNodeName). Yes, it is better.

**Pull request merged**